src/compute-client/src/controller.rs

```diff
@@ -144,7 +145,7 @@ impl ComputeReplicaLogging {
 }
 
 /// A controller for the compute layer.
-pub struct ComputeController<T> {
+pub struct ComputeController<T: Timestamp> {
```

Comment:
`ReadHold<T>` includes this bound in the struct definition (it has to because its `Drop` impl requires the `Timestamp` bound), so every type that stores a `ReadHold` also needs to have the bound. You'll see this a bunch in this PR.